naxsi_src/naxsi_runtime.c

```diff
@@ -806,9 +806,11 @@ ngx_int_t ngx_http_nx_log(ngx_http_request_ctx_t *ctx,
   u_int		sz_left, sub, offset = 0, i;
   ngx_str_t	*fragment, *tmp_uri;
   ngx_http_special_score_t	*sc;
-  const char 	*fmt_base = "ip=%.*s&server=%.*s&uri=%.*s&learning=%d&vers=%.*s&total_processed=%zu&total_blocked=%zu&block=%d";
+  const char 	*fmt_base = "ip=%.*s&server=%.*s&uri=%.*s&vers=%.*s&total_processed=%zu&total_blocked=%zu&config=%.*s";
   const char	*fmt_score = "&cscore%d=%.*s&score%d=%zu";
   const char	*fmt_rm = "&zone%d=%s&id%d=%d&var_name%d=%.*s";
+  const char    *fmt_config =  ctx->learning ? (ctx->drop ? "learning-drop" : "learning" ) :  (ctx->block ? "block" : (ctx->drop ? "drop" : ""));
```

Comment:
I think we want to be able to distinguish a request that was blocked because a rule trigger a `drop` while `learning` mode is off from a request that trigger a `block` while `learning` mode is off  :)

(I tend to use DROP rules for vpatching or so, and thus might want to react differently to blocked and dropped requests)

Reply:
This is the case in my proposal, or am I missing something ?

The only quirk is if the request is blockcked and dropped, it will be seen as blocked as it is evaluated first...

Reply:
Yes, this is my point :) I guess in terms of priority drop > block

Reply:
]ok, then. It makes sense.

b94415f fixes this.